cmd/dp-import-reporter/main.go

```diff
@@ -0,0 +1,100 @@
+package main
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"runtime/debug"
+	"syscall"
+
+	"github.com/ONSdigital/dp-import-reporter/config"
+	"github.com/ONSdigital/go-ns/kafka"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/coocood/freecache"
+	"github.com/ONSdigital/dp-import-reporter/event"
+	"github.com/ONSdigital/dp-import-reporter/client"
+	"net/http"
+	"io/ioutil"
+	"io"
+	"github.com/ONSdigital/dp-import-reporter/server"
+	"github.com/ONSdigital/dp-import-reporter/message"
+)
+
+func main() {
+	log.Namespace = "dp-event-reporter"
+
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, os.Interrupt, syscall.SIGTERM)
+	errorChannel := make(chan error, 1)
+
+	cfg, err := config.Get()
+	if err != nil {
+		os.Exit(1)
+	}
+
+	log.Info("dp-import-reporter config", log.Data{
+		"config": cfg,
+	})
+
+	datasetAPIClient, err := client.NewDatasetAPIClient(cfg.DatasetAPIURL, cfg.DatasetAPIAuthToken, &http.Client{}, ResponseBodyReader{})
+	if err != nil {
+		os.Exit(1)
+	}
+
+	cache := freecache.NewCache(cfg.CacheSize)
+	// TODO why is this set?
+	debug.SetGCPercent(20)
+
+	server.Start(cache, cfg.BindAddress, errorChannel)
+
+	reportEventHandler := event.Handler{
+		Cache:         cache,
+		DatasetAPI:    datasetAPIClient,
+		ExpireSeconds: cfg.CacheExpiry,
+	}
+
+	eventReceiver := &event.Receiver{
+		Handler: reportEventHandler,
+	}
+
+	// create the report event kafka kafkaConsumer.
+	kafkaConsumer, err := kafka.NewConsumerGroup(cfg.Brokers, cfg.ReportEventTopic, log.Namespace, kafka.OffsetNewest)
```

Comment:
Will we always want to consume off the newest message in our kafka queue? Would we want the ability to replay  a message? To answer my own question... I would say no until there is a case for implementing a configurable offset and the default will always be `OffsetNewest`.